include/fw_info.h

```diff
@@ -248,51 +348,19 @@ static inline const struct fw_info *fw_info_find(u32_t firmware_address)
 }
 
 
-/* Check a fw_info_ext_api pointer. */
-static inline bool fw_info_ext_api_check(
-				const struct fw_info_ext_api *ext_api_info)
-{
-	const u32_t ext_api_info_magic[] = {EXT_API_MAGIC};
-
-	return memcmp(ext_api_info->magic, ext_api_info_magic,
-			CONFIG_FW_INFO_MAGIC_LEN) == 0;
-}
-
-
 /**Expose EXT_APIs to another firmware
  *
  * Populate the other firmware's @c ext_api_in with EXT_APIs from other images.
  *
  * @note This is should be called immediately before booting the other firmware
  *       since it will likely corrupt the memory of the running firmware.
  *
- * @param[in]  fwinfo  Pointer to the other firmware's information structure.
- */
-void fw_info_ext_api_provide(const struct fw_info *fwinfo);
-
-/**Get a single EXT_API.
- *
- * @param[in]    id      Which EXT_API to get.
- * @param[in]    index   If there are multiple EXT_APIs available with the same
- *                       ID, retrieve the different ones with this.
- *
- * @return The EXT_API, or NULL, if it wasn't found.
- */
-const struct fw_info_ext_api *fw_info_ext_api_get(u32_t id, u32_t index);
-
-/**Find an EXT_API based on a version range.
- *
- * @param[in]  id           The ID of the EXT_API to find.
- * @param[in]  flags        The required flags of the EXT_API to find. The
- *                          returned EXT_API may have other flags set as well.
- * @param[in]  min_version  The minimum acceptable EXT_API version.
- * @param[in]  max_version  One more than the maximum acceptable EXT_API
- *                          version.
+ * @param[in]  fw_info    Pointer to the other firmware's information structure.
+ * @param[in]  test_only  Don't populate ext_api_in.
  *
- * @return The EXT_API, or NULL if none was found.
+ * @return Whether requirements could be satisified.
  */
-const struct fw_info_ext_api *fw_info_ext_api_find(u32_t id, u32_t flags,
-					u32_t min_version, u32_t max_version);
+bool fw_info_ext_api_provide(const struct fw_info *fwinfo, bool test_only);
```

Comment:
I don't like having code for testing only. Could this be handled in a different way? e.g. macro, creating a wrapper or something?

Reply:
It refers to testing for the existence of the ext_apis. To be clear, the argument is meant to be used in regular code, not in tests. We can probably give it a better name.

Reply:
I only assumed that it was only used for testing.

To me "test only" heavily implies that this is only used for testing, but perhaps that is just me :) Up to you if you want to rename it.

Reply:
I'll try to find another name :)

Reply:
Done (`provide`)